### PR TITLE
Fix of the theoretical race condition in case of multiple calls to JITM cache

### LINKED ISSUE #12529: IllegalStateException: Already resumed
Sentry Issue: [WOOCOMMERCE-ANDROID-CK3](https://a8c.sentry.io/issues/5674827212/?referrer=github_integration)

```
IllegalStateException: Already resumed
    at com.woocommerce.android.ui.jitm.JitmStoreInMemoryCache.init(JitmStoreInMemoryCache.kt:57)
    cacheInitContinuation?.resume(Unit)
    at com.woocommerce.android.ui.jitm.JitmStoreInMemoryCache$init$1.invokeSuspend(JitmStoreInMemoryCache.kt:0)
...
(9 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #12529

### Description

My theory is that this started to happen as JITM was placed on the Store screen, so in some cases, we may get both `init` and `getMessagesForPath` roughly in parallel, which causes continuation to be resumed twice

I could not reproduce this even with a test, but I could reproduce a deadlock with this test. 

### Steps to reproduce

I could not reproduce this

### Testing information

Maybe try to reproduce this with another unit test?

### The tests that have been performed

Only unit test and I simulated JITM to be shown

### Images/gif

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/234">20.3</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by samiuelson — APPROVED** [review]
LGTM!

I'm leaving the PR not merged in case anyone else wants to review or can reproduce the issue.

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/jitm/JitmStoreInMemoryCacheTest.kt [hunk 1]
```diff
@@ -213,4 +217,47 @@ class JitmStoreInMemoryCacheTest : BaseUnitTest() {
         // THEN
         assertThat(cache.getMessagesForPath(messagePath = messagePath)).isEmpty()
     }
+
+    @Test
+    fun `given concurrent calls to getMessagesForPath, when init is not complete, then IllegalStateException is thrown`() =
```
Comment (AnirudhBhat) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/jitm/JitmStoreInMemoryCacheTest.kt]: ❓ Should the test name be `IllegalStateException is not thrown`?

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/jitm/JitmStoreInMemoryCacheTest.kt [hunk 2]
```diff
@@ -213,4 +217,47 @@ class JitmStoreInMemoryCacheTest : BaseUnitTest() {
...
+                    } catch (e: Exception) {
+                        exceptions.add(e)
+                    }
+                }
+            }
+
+            val jobsGetMessages = List(50) {
+                async(Dispatchers.Default) {
+                    try {
+                        cache.init()
```
Comment (AnirudhBhat) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/jitm/JitmStoreInMemoryCacheTest.kt]: ❓ Should this be `cache.getMessagesForPath`?

Comment (kidinov) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/jitm/JitmStoreInMemoryCacheTest.kt]: Good catch - fixed here eadf4269915c3e9d370947ce85ef55673faab9d8

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/jitm/JitmStoreInMemoryCacheTest.kt (hunk 1 above)
Comment (kidinov) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/jitm/JitmStoreInMemoryCacheTest.kt]: Good catch - fixed here eadf4269915c3e9d370947ce85ef55673faab9d8

**Comment by backwardstruck** [comment]
I think we can probably check these ones off right?

```
Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
```


I know I also forget to mark them as done 😸

**Pull request merged**